r/inst/build_arrow_static.sh

```diff
@@ -47,6 +47,12 @@ else
   ARROW_DEFAULT_PARAM="OFF"
 fi
 
+# Snappy 1.1.9 is patched to implement https://github.com/google/snappy/pull/148 but some platforms don't have
+# patch available, so disable snappy in those cases. If the snappy version is bumped, we should remove this.
+if [ ! $(command -v patch) ]; then
+  ARROW_WITH_SNAPPY=OFF
+fi
```

Comment:
For our CI this should not be hit, but is important incase CRAN happens to not have `patch`, it will disable snappy so we don't get a sanitizer error there.